### ISSUE TITLE
Not working on the latest MacOS 12.3

### ISSUE THREAD

**Issue by wisinfun** [deanishe/alfred-workflow#182]
First of all, thank you.
Not working on the latest MacOS 12.3

**Comment by fedecalendino** [comment]
I think Apple finally removed python2.7 from macOS since the command `python` cannot be found by Alfred.

~~edit: Migrating the workflow library with `2to3` and using `python3` in my Script Filters, did the trick to revive my workflows: https://github.com/fedecalendino/alfred-workflow/commit/650e193cda80e3fe16cb7fcbfa805e71371f6d93~~

~~Although they are quite simple, so it might not work for everyone as well.~~

**Comment by psifertex** [comment]
It's especially ironic given the text:

```
Alfred-Workflow is targeted at the system Python on macOS. Its goal is to enable developers to build workflows that will “just work” for users on any vanilla installation of macOS since Snow Leopard.
```

Yeah, well, for some time it's been obviously that Python2 was [going away](https://pythonclock.org/). For it to "just work" now, it _needs_ python 3 support. 

at the bottom of https://www.deanishe.net/alfred-workflow/supported-versions.html#why-no-python-3-support

**Comment by psifertex** [comment]
For anyone else in the same boat, might want to check out this fork: https://github.com/NorthIsUp/alfred-workflow-py3

(found via the pinned issue: https://github.com/deanishe/alfred-workflow/issues/97)

**Comment by MattWenJun** [comment]
macOS has removed the python2.7. You need to follow the steps to solve your problem:

First, you need to reinstall Python 2. To do so, install Homebrew then run the following in a terminal:
```
export PATH="/opt/homebrew/bin:/usr/local/bin:${PATH}"
eval "$(brew shellenv)"
brew install pyenv
pyenv install 2.7.18
ln -s "${HOME}/.pyenv/versions/2.7.18/bin/python2.7" "${HOMEBREW_PREFIX}/bin/python"
```
Now you can use "python filename.py" to run your python2 file in the macOS terminal.
You can also run your python2 code in the Alfred  Script Filter.

But if you use bash to run your python2 file in Alfred, you need to do one more thing.
Change your bash command in your workflow configuration from：
`python filename.py "{query}"`
to：
`${HOME}/.pyenv/versions/2.7.18/bin/python filename.py "{query}"`
If you don't do this you will get an error report by Alfred debugger which is "/bin/bash: python: command not found"

**Comment by xavdid** [comment]
Here's Apple's [release notes](https://developer.apple.com/documentation/macos-release-notes/macos-12_3-release-notes#Python) on the subject. Namely:

> Python 2.7 was removed from macOS in this update. Developers should use Python 3 or an alternative language instead.

See also, [this Stack Exchange thread](https://apple.stackexchange.com/a/436059/398530)

**Comment by FischLu** [comment]
> It's especially ironic given the text:
> 
> ```
> Alfred-Workflow is targeted at the system Python on macOS. Its goal is to enable developers to build workflows that will “just work” for users on any vanilla installation of macOS since Snow Leopard.
> ```
> 
> Yeah, well, for some time it's been obviously that Python2 was [going away](https://pythonclock.org/). For it to "just work" now, it _needs_ python 3 support.
> 
> at the bottom of https://www.deanishe.net/alfred-workflow/supported-versions.html#why-no-python-3-support

I agree with you, the sentence you quoted is ridiculous. Even if the developers stick to their views, they should be prepared for python3 in another branch. "just work" can become "just not work" in one night, now, it happens and no official "just work" solutions.

**Comment by raingao** [comment]
> macOS has removed the python2.7. You need to follow the steps to solve your problem:
> 
> First, you need to reinstall Python 2. To do so, install Homebrew then run the following in a terminal:
> 
> ```
> export PATH="/opt/homebrew/bin:/usr/local/bin:${PATH}"
> eval "$(brew shellenv)"
> brew install pyenv
> pyenv install 2.7.18
> ln -s "${HOME}/.pyenv/versions/2.7.18/bin/python2.7" "${HOMEBREW_PREFIX}/bin/python"
> ```
> 
> Now you can use "python filename.py" to run your python2 file in the macOS terminal. You can also run your python2 code in the Alfred Script Filter.
> 
> But if you use bash to run your python2 file in Alfred, you need to do one more thing. Change your bash command in your workflow configuration from： `python filename.py "{query}"` to： `${HOME}/.pyenv/versions/2.7.18/bin/python filename.py "{query}"` If you don't do this you will get an error report by Alfred debugger which is "/bin/bash: python: command not found"

and you should replace '/usr/bin/python' -> '/usr/local/bin/python' in /workflow/background.py

**Comment by fedecalendino** [comment]
I also don't get why we didn't have a version of the code that would work on both version of Python, plenty of libraries do that.

**Comment by techouse** [comment]
I faced the same issue a few weeks ago and decided to bite the bullet and migrate all my workflows to Dart [because it can be compiled into a standalone executable](https://dart.dev/tools/dart-compile) eliminating the need for any external prerequisites.

In the process, I had to build [a library just like this one in Dart](https://github.com/techouse/alfred_workflow). The [Alfred docs](https://www.alfredapp.com/help/workflows/inputs/script-filter/json/) helped a lot. Was a lot of work but fun 😊

**Comment by devnoname120** [comment]
See https://github.com/deanishe/alfred-workflow/pull/183

**Comment by xieao100** [comment]
> > macOS 已移除 python2.7。您需要按照以下步骤解决您的问题：
> > 首先，您需要重新安装 Python 2。为此，请安装 Homebrew，然后在终端中运行以下命令：
> > ```
> > export PATH="/opt/homebrew/bin:/usr/local/bin:${PATH}"
> > eval "$(brew shellenv)"
> > brew install pyenv
> > pyenv install 2.7.18
> > ln -s "${HOME}/.pyenv/versions/2.7.18/bin/python2.7" "${HOMEBREW_PREFIX}/bin/python"
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > 现在您可以使用“python filename.py”在 macOS 终端中运行您的 python2 文件。您还可以在 Alfred 脚本过滤器中运行您的 python2 代码。
> > 但是如果你使用 bash 在 Alfred 中运行你的 python2 文件，你还需要做一件事。将工作流配置中的 bash 命令从：`python filename.py "{query}"`更改为：`${HOME}/.pyenv/versions/2.7.18/bin/python filename.py "{query}"`如果您不这样做，您将收到 Alfred 调试器的错误报告，即“/bin/bash: python: command not found”
> 
> 你应该在 /workflow/background.py 中替换 '/usr/bin/python' -> '/usr/local/bin/python'

非常感谢！解决了我的问题！

**Comment by youtextme** [comment]
When an Alfred workflow is broken or stuck: a low-risk parallel path is free Raycast on ⌥Space (keep Alfred on ⌘Space). Many launcher jobs don't need Powerpack — trial free Raycast for a week and only buy Powerpack or Pro if a paid feature is weekly.

https://raycast.com/?via=girish
Affiliate — I earn a commission if you upgrade to Pro.
